plugins/coverbrowser.koplugin/bookinfomanager.lua

```diff
@@ -626,6 +626,245 @@ function BookInfoManager:cleanUp()
+                end
+            else
+                UIManager:close(info)
+                info = InfoMessage:new{text = summary}
+                UIManager:show(info)
+                UIManager:forceRePaint()
+                util.sleep(2) -- Let the user see that
+                break
+            end
+        end
+        UIManager:close(info)
+    end
+
+    local files
+    while true do
+        info = InfoMessage:new{text = _("Looking for books to index…")}
+        UIManager:show(info)
+        UIManager:forceRePaint()
+        completed, files = Trapper:dismissableRunInSubprocess(function()
+            local files = findFilesInDir(path, recursive)
+            table.sort(files)
+            return files
+        end, info)
+        if not completed then
+            abort = Trapper:confirm(_("Do you want to abort extraction?"), _("No"), _("Yes"))
+            if abort then
+                return
+            end
+        elseif not files or #files == 0 then
+            UIManager:close(info)
+            info = InfoMessage:new{text = _("No book were found.")}
+            UIManager:show(info)
+            return
+        else
+            break
+        end
+    end
+    UIManager:close(info)
+
+    if refresh_existing then
+        info = InfoMessage:new{text = T(_("Found %1 books to index."), #files)}
+        UIManager:show(info)
+        UIManager:forceRePaint()
+        util.sleep(2) -- Let the user see that
+    else
+        local all_files = files
+        while true do
+            info = InfoMessage:new{text = T(_("Found %1 books.\nLooking for those not already present in cache database…"), #all_files)}
+            UIManager:show(info)
+            UIManager:forceRePaint()
+            util.sleep(2) -- Let the user see that
+            completed, files = Trapper:dismissableRunInSubprocess(function()
+                files = {}
+                for _, filepath in pairs(all_files) do
+                    local bookinfo = self:getBookInfo(filepath)
+                    local to_extract = not bookinfo
+                    if bookinfo and cover_specs and not bookinfo.ignore_cover then
+                        if bookinfo.cover_fetched then
+                            if bookinfo.has_cover and cover_specs.sizetag ~= bookinfo.cover_sizetag then
+                                if bookinfo.cover_sizetag ~= "M" then -- keep the bigger "M"
+                                    to_extract = true
+                                end
+                            end
+                        else
+                            to_extract = true
+                        end
+                    end
+                    if to_extract then
+                        table.insert(files, filepath)
+                    end
+                end
+                return files
+            end, info)
+            if not completed then
+                abort = Trapper:confirm(_("Do you want to abort extraction?"), _("No"), _("Yes"))
```

Comment:
Don't abort, Abort?

Reply:
Continue / Abort ?

Reply:
Right, better.

Reply:
Although it might be unclear what continue is.

Reply:
Yes, thought so too. Will go with "Don't abort".